backend/apid/routers/apikeys.go

```diff
@@ -0,0 +1,101 @@
+package routers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	corev2 "github.com/sensu/sensu-go/api/core/v2"
+	"github.com/sensu/sensu-go/backend/apid/actions"
+	"github.com/sensu/sensu-go/backend/apid/handlers"
+	"github.com/sensu/sensu-go/backend/store"
+)
+
+// APIKeysRouter handles requests for /apikeys.
+type APIKeysRouter struct {
+	handlers handlers.Handlers
+}
+
+// NewAPIKeysRouter instantiates new router for controlling apikeys resources.
+func NewAPIKeysRouter(store store.ResourceStore) *APIKeysRouter {
+	return &APIKeysRouter{
+		handlers: handlers.Handlers{
+			Resource: &corev2.APIKey{},
+			Store:    store,
+		},
+	}
+}
+
+// Mount the APIKeysRouter to a parent Router.
+func (r *APIKeysRouter) Mount(parent *mux.Router) {
+	routes := ResourceRoute{
+		Router:     parent,
+		PathPrefix: "/{resource:apikeys}",
+	}
+
+	routes.Del(r.handlers.DeleteResource)
+	routes.Get(r.handlers.GetResource)
+	routes.List(r.handlers.ListResources, corev2.APIKeyFields)
+	routes.Patch(r.update)
+	parent.HandleFunc(routes.PathPrefix, r.create).Methods(http.MethodPost)
+}
+
+func (r *APIKeysRouter) create(w http.ResponseWriter, req *http.Request) {
+	apikey := &corev2.APIKey{}
+	if err := UnmarshalBody(req, apikey); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// set/overwrite the key id and created_at time
+	key, err := uuid.NewRandom()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	apikey.Name = key.String()
+	apikey.CreatedAt = time.Now().Unix()
+	newBytes, err := json.Marshal(apikey)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	req.Body = ioutil.NopCloser(bytes.NewBuffer(newBytes))
+
+	_, err = r.handlers.CreateResource(req)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// set the relative location header
+	w.Header().Set("Location", fmt.Sprintf("%s/%s", req.URL.String(), apikey.Name))
+	w.WriteHeader(http.StatusCreated)
+}
+
+func (r *APIKeysRouter) update(req *http.Request) (interface{}, error) {
```

Comment:
Stupid question: The specs mentions we should support updating an API key to update its binding. Considering we really only store the username in the key now, should we still support updating an API key?

Reply:
I've removed the functionality to update an api key. If we decide we want to support groups (and updating them) in the future, we can reimplement.